### PR TITLE
Stop using latest as the default image tag for OA and AG

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

We will always set the `Status.<Component>.ImageID` to a specific version, even in the default case.
If a custom version is provided, we will append the `raw` postfix if it is not present.

Minor change in the behavior of the troubleshooting script: 
- Now we have to rely on the Status, otherwise, we would need to get the latest versions there as well.
- I also removed some tests, as they were totally unnecessary and sometimes even incorrect

## How can this be tested?

When deploying OA or AG components, the version is "always" (except with custom images) in the image tag.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: pkg/util/dtversion/dtversion.go [hunk 1]
```diff
@@ -48,3 +50,14 @@ func ToSemver(version string) (string, error) {
 
 	return result, nil
 }
+
+func ToImageTag(version string) string {
```
Comment (andriisoldatenko) [inline pkg/util/dtversion/dtversion.go:54]: @0sewa0 could you please add unit-test to this function?

Comment (0sewa0) [reply pkg/util/dtversion/dtversion.go:54]: Question: This function is rather dumb, as it doesn't do any verification, should I add a verification?

The problem with that is that I would need to modify the signature of the function, and the signature of most functions that use it. (shouldn't be that problematic)

Comment (andriisoldatenko) [reply pkg/util/dtversion/dtversion.go:54]: you can remove/refactor or change it :) just noticed it doesn't have tests...

File: cmd/troubleshoot/dynakube_test.go [hunk 2]
```diff
@@ -349,10 +358,12 @@ func (builder *testDynaKubeBuilder) withCloudNativeFullStackCustomImage(image st
 func (builder *testDynaKubeBuilder) withHostMonitoringCustomImage(image string) *testDynaKubeBuilder {
 	if builder.dynakube.Spec.OneAgent.HostMonitoring != nil {
 		builder.dynakube.Spec.OneAgent.HostMonitoring.Image = image
+		builder.dynakube.Status.OneAgent.ImageID = image
```
Comment (andriisoldatenko) [inline cmd/troubleshoot/dynakube_test.go:361]: from what I can see you can do this line out of if/else. cause we do same in both branches. 🤔

File: cmd/troubleshoot/dynakube_test.go [hunk 3]
```diff
@@ -361,13 +372,15 @@ func (builder *testDynaKubeBuilder) withHostMonitoringCustomImage(image string)
...
 		builder.dynakube.Spec.OneAgent.CloudNativeFullStack.CodeModulesImage = image
+		builder.dynakube.Status.CodeModules.ImageID = image
 	} else {
 		builder.dynakube.Spec.OneAgent.CloudNativeFullStack = &dynatracev1beta1.CloudNativeFullStackSpec{
 			AppInjectionSpec: dynatracev1beta1.AppInjectionSpec{
 				InitResources:    &corev1.ResourceRequirements{},
 				CodeModulesImage: image,
 			},
 		}
+		builder.dynakube.Status.CodeModules.ImageID = image
```
Comment (andriisoldatenko) [inline cmd/troubleshoot/dynakube_test.go:383]: same here.

File: cmd/troubleshoot/dynakube_test.go [hunk 4]
```diff
@@ -335,12 +342,14 @@ func (builder *testDynaKubeBuilder) withClassicFullStackCustomImage(image string
 func (builder *testDynaKubeBuilder) withCloudNativeFullStackCustomImage(image string) *testDynaKubeBuilder {
 	if builder.dynakube.Spec.OneAgent.CloudNativeFullStack != nil {
 		builder.dynakube.Spec.OneAgent.CloudNativeFullStack.Image = image
+		builder.dynakube.Status.OneAgent.ImageID = image
```
Comment (andriisoldatenko) [inline cmd/troubleshoot/dynakube_test.go:345]: same here

File: cmd/troubleshoot/dynakube_test.go [hunk 5]
```diff
@@ -304,29 +306,34 @@ func (builder *testDynaKubeBuilder) withCloudNativeFullStack() *testDynaKubeBuil
...
 	builder.dynakube.Spec.OneAgent.HostMonitoring = &dynatracev1beta1.HostInjectSpec{}
+	builder.dynakube.Status.OneAgent.ImageID = builder.dynakube.DefaultOneAgentImage(testVersion)
 
 	return builder
 }
 
 func (builder *testDynaKubeBuilder) withClassicFullStackCustomImage(image string) *testDynaKubeBuilder {
 	if builder.dynakube.Spec.OneAgent.ClassicFullStack != nil {
 		builder.dynakube.Spec.OneAgent.ClassicFullStack.Image = image
+		builder.dynakube.Status.OneAgent.ImageID = image
```
Comment (andriisoldatenko) [inline cmd/troubleshoot/dynakube_test.go:331]: and here :)

File: config/crd/bases/dynatrace.com_dynakubes.yaml [hunk 6]
```diff
@@ -3667,10 +3667,6 @@ spec:
                       performed
                     format: date-time
                     type: string
-                  lastProcessModuleConfigUpdate:
```
Comment (andriisoldatenko) [inline config/crd/bases/dynatrace.com_dynakubes.yaml:3670]: you need rebase :)

Comment (0sewa0) [reply config/crd/bases/dynatrace.com_dynakubes.yaml:3670]: github seems to think its fine :D

Comment (andriisoldatenko) [reply config/crd/bases/dynatrace.com_dynakubes.yaml:3670]: image

File: cmd/troubleshoot/dynakube_test.go (hunk 4 above)
Comment (0sewa0) [reply cmd/troubleshoot/dynakube_test.go]: done

File: cmd/troubleshoot/dynakube_test.go (hunk 3 above)
Comment (0sewa0) [reply cmd/troubleshoot/dynakube_test.go]: doen

File: pkg/util/dtversion/dtversion.go (hunk 1 above)
Comment (0sewa0) [reply pkg/util/dtversion/dtversion.go:54]: done

**Pull request merged**